src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -298,15 +298,16 @@ private void AddClassifiedSpans(IEditorClassificationService classificationServi
             }
 
             private void AddClassifiedSpansForCurrentTree(
-                IEditorClassificationService classificationService, SnapshotSpan span, Document document, List<ClassifiedSpan> classifiedSpans)
+                ClassificationService classificationService, SnapshotSpan span, Document document, List<ClassifiedSpan> classifiedSpans)
             {
                 List<ClassifiedSpan> tempList;
                 if (!_lastLineCache.TryUseCache(span, out tempList))
                 {
                     tempList = ClassificationUtilities.GetOrCreateClassifiedSpanList();
 
-                    classificationService.AddSyntacticClassificationsAsync(
-                        document, span.Span.ToTextSpan(), tempList, CancellationToken.None).Wait(CancellationToken.None);
+                    tempList.AddRange(
+                        classificationService.GetSyntacticClassificationsAsync(
```

Comment:
This undoes the work of reusing hte list.  presumably GetSyntacticClassificationsAsync will now allocate.  And that was something we were trying to avoid before.  Can we not have the pattern where the list (or context) is passed in?